power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -45,8 +47,122 @@ struct DatasetInfo {
     Idx batch_size; // for single dataset, the batch size is one
     MetaDataset const* dataset;
     std::vector<ComponentInfo> component_info;
+
+    DatasetInfo(bool is_batch, Idx batch_size, MetaDataset const* dataset)
```

Comment:
Why creating a constructor? This type is not aggregate anymore and we cannot use the named attribute initialization.

Reply:
Addressed: Made ComponentInfo and DatasetInfo aggregate with default values.